# 2: Determinate Beams and Frames
## 2.5: Beam Reactions Example

Consider the following beam structure, below which is shown the overall FBD.

![FBD 0](../../images/sdbeams/fbd/drawing-1.svg)

There are 4 unknown reactions - the 3 available equations of equilibrium are not sufficient to solve for the reactions.

### 2.5.1: Free Body Diagram of Part b-c-d-e

The pin at point $b$ requires the internal bending moment at that point to be 0.  That allows us to
split the structure at that spot to develop an additional FBD.  It only adds 2 additional unknowns
(horizontal and vertical force at $b$) and so the three additional equilibrium equations will now
be sufficient.

We choose a side that involves only one unknown in an equation, allowing us to solve for
that unknown immediately; here the right side:

![FBD 1](../../images/sdbeams/fbd/drawing-2.svg)

In [1]:
from sympy import symbols,solve
from sympy.physics.units import m,newton,force,kilo,Quantity
kilonewton = kilonewtons = kN = Quantity("kilonewton", "kN")
kN.set_dimension(force)
kN.set_scale_factor(kilo*newton)

In [2]:
Hb,Vb,Vd = symbols('H_b V_b V_d')

### $\sum M_b = 0~~~~~+ccw$

In [3]:
sum_Mb = -65*kN*4*m  - 28*kN/m*(6*m+2*m)*(6*m+2*m)/2 + Vd*6*m
sum_Mb

-1156*kilonewton*meter + 6*meter*V_d

In [4]:
Vd = solve(sum_Mb,Vd)[0].n()
Vd

192.666666666667*kilonewton

### $\sum F_y = 0~~~~~+\uparrow$

In [5]:
sum_Fy = Vb - 28*kN/m*(6*m+2*m) - 65*kN + Vd
sum_Fy

-96.3333333333333*kilonewton + V_b

In [6]:
Vb = solve(sum_Fy,Vb)[0].n()
Vb

96.3333333333333*kilonewton

### $\sum F_x = 0~~~~~+ \rightarrow$

In [7]:
sum_Fx = Hb
Hb = solve(sum_Fx,Hb)[0].n()
Hb

0

In [8]:
Vd/kN, Vb/kN, Hb/kN

(192.666666666667, 96.3333333333333, 0)

### 2.5.2: Free Body Diagram of part a-b
![FBD a-b](../../images/sdbeams/fbd/drawing-3.svg)

In [9]:
Ha,Ma,Va = symbols('H_a M_a V_a')

### $\sum F_x = 0~~~~~+\rightarrow$

In [10]:
sum_Fx = Ha - Hb
Ha = solve(sum_Fx,Ha)[0].n()
Ha

0

### $\sum F_y = 0~~~~~+\uparrow$

In [11]:
sum_Fy = Va - 28*(kN/m)*8*m - Vb
Va = solve(sum_Fy,Va)[0].n()
Va

320.333333333333*kilonewton

### $\sum M_a = 0~~~~~+ ccw$

In [12]:
sum_Ma = Ma - 28*(kN/m)*8*m*8*m/2 - Vb*8*m
Ma = solve(sum_Ma,Ma)[0].n()
Ma

1666.66666666667*kilonewton*meter

### 2.5.3: Equilibrium Check

We can use a third FBD as an equilibrium check.  It is not an independent FBD but it will at
least check for some numerical errors, giving us some confidence that the work is correct.

![Entire Beam FBD](../../images/sdbeams/fbd/drawing-4.svg)

### $\sum F_x~~~~~+ \rightarrow$

In [13]:
Ha

0

### $\sum F_y~~~~~+ \uparrow$

In [14]:
Va + Vd - 28*(kN/m)*(14*m+2*m) - 65*kN

0

### $\sum M_e~~~~~+ccw$

In [15]:
Ma + 28*(kN/m)*(14*m+2*m)*(14*m+2*m)/2 + 65*kN*(2*m+2*m) - Va*(14*m+2*m) - Vd*2*m

6.25277607468888e-13*kilonewton*meter

$6.25 \times 10^{-13}$ is easily close enough to zero, attributable to normal floating point truncation errors.

**Note:** If you were using manual calculations and the numeric valies of $M_a$, $V_a$ and $V_d$ rounded to 4 significant
figures, this last equilibrium check would give something quite a bit further from zero, but still acceptable.